torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,513 @@
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.ids)
+
+    def _load_raster(self, filename: int, source: str) -> Tensor:
+        """Load a single raster image or target.
+
+        Args:
+            filename: name of the file to load
+            source: the filename of the modality
+
+        Returns:
+            the raster image or target
+        """
+        with rasterio.open(
+            os.path.join(self.root, f"{source}", f"{filename}.tif")
+        ) as f:
+            array = f.read()
+            if array.dtype == np.uint16:
+                array = array.astype(np.int32)
+            tensor = torch.from_numpy(array)
+            return tensor
+
+    def _verify(self, url: str, md5: str) -> None:
+        """Verify the integrity of the dataset."""
+        # Check if the files already exist
+        if os.path.exists(os.path.join(self.root, url.split("/")[-2])):
+            return
+
+        # Check if the zip file has already been downloaded
+        pathname = os.path.join(self.root, os.path.split(url)[1].split(".")[0])
+        if os.path.exists(pathname):
+            self._extract(url)
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                f"Dataset not found in `root={self.root}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
+        else:
+            # Download the dataset
+            self._download(url, md5)
+            self._extract(url)
+
+    def _verify_split(self) -> None:
+        """Verify the integrity of the split file."""
+        # Download the dataset
+        download_url(
+            self.split_url,
+            self.root,
+            filename=os.path.split(self.split_url)[1],
+            md5=None,
+        )
+
+    def _download(self, url: str, md5: str) -> None:
+        """Download the dataset."""
+        download_url(
+            url,
+            self.root,
+            filename=os.path.split(url)[1],
+            md5=md5 if self.checksum else None,
+        )
+
+    def _extract(self, url: str) -> None:
+        """Extract the dataset."""
+        filepath = os.path.join(self.root, os.path.split(url)[1])
+        extract_archive(filepath)
+
+    def _check_integrity(self, filename: str, md5: str) -> bool:
+        """Check integrity of dataset.
+
+        Returns:
+            True if dataset files are found and/or MD5s match, else False
+        """
+        filepath = os.path.join(self.root, os.path.split(filename)[1])
+        if not check_integrity(filepath, md5 if self.checksum else None):
+            return False
+        return True
+
+    def merge_parts(self, source_path: str, modality: str) -> None:
```

Comment:
Could probably start with an underscore to make it hidden. Need to document parameters